ql/ql/src/queries/style/CountingToZero.ql

```diff
@@ -0,0 +1,21 @@
+/**
+ * @name Counting zero elements
+ * @description Use not exists instead of checking that there is zero elements in a set.
+ * @kind problem
+ * @problem.severity warning
+ * @id ql/counting-to-zero
+ * @precision high
+ */
+
+import ql
+
+from ComparisonFormula comp, int c, string msg
+where
+  c = 0 and // [0, 1] and // skipping the 1 case for now, as it seems too noisy.
+  comp.getOperator() = ["=", "!="] and
+  comp.getAnOperand().(Integer).getValue() = c and
+  comp.getAnOperand().(Aggregate).getKind() = ["count", "strictcount"] and
```

Comment:
This is probably not really an issue, but I'm gonna write it up anyway: If a predicate is tagged with `language[monotonicAggregates]` I guess `count/strictcount` and `not exists` have different monotonicity properties. But I guess there's no harm in surfacing the alert so that the author can think about whether it's possible to replace the aggregate with a `not exist`.

Reply:
> But I guess there's no harm in surfacing the alert so that the author can think about whether it's possible to replace the aggregate with a not exist. 

I think so too.  
I'm not currently making an automated patch out of this yet. If I did then I would have to consider such cases.